src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -344,4 +347,76 @@ public Graph getVocabularyMappingAsGraph() {
 		return vocabularyMapping;
 	}
 
+	@Override
+	public Set<SolutionMapping> translateSolutionMapping( final SolutionMapping sm ) {		
+		Set<BindingBuilder> bbs = new HashSet<>();
+		bbs.add( BindingBuilder.create() );
+		
+		final Iterator<Var> i = sm.asJenaBinding().vars();
+		while(i.hasNext()) {
+			final Var v = i.next();
+			final Node n = sm.asJenaBinding().get(v);
+			
+			if(!n.isURI()) {
+				continue;
+			}
+			
+			final Set<Node> bindingTranslation = translateBinding(n);
+			if (bindingTranslation.size() > 1) {
+				final Set<BindingBuilder> bbsCopy = new HashSet<>();
+				
+				for(final Node j : bindingTranslation) {
+					for (final BindingBuilder k : bbs) {
+						BindingBuilder translationCopy = BindingBuilder.create();
+						if(!k.isEmpty()) {
+							translationCopy.addAll(k.snapshot());
+						}
+						translationCopy.add(v, j);
+						bbsCopy.add(translationCopy);
+					}
+				}
+				
+				bbs = bbsCopy;
+			} else if (bindingTranslation.size() == 0) {
+				continue;
+			} else {
```

Comment:
Same issue here. I think that simply doing `continue` here will resulting in losing the binding for the current variable.

Also here, please cover this case in the additional unit test.